noxfile.py

```diff
@@ -108,3 +108,52 @@ def docs(session):
         sphinx_args.insert(0, "--open-browser")
 
     session.run(sphinx_cmd, *sphinx_args)
+
+
+# The following sessions are only to be run in CI to check the nox GHA action
+def _check_python_version(session):
+    if session.python.startswith("pypy"):
+        python_version = session.python[4:]
+        implementation = "pypy"
+    else:
+        python_version = session.python
+        implementation = "cpython"
+    session.run(
+        "python",
+        "-c",
+        "import sys; assert '.'.join(str(v) for v in sys.version_info[:2]) =="
+        f" '{python_version}'",
+    )
+    if python_version[:2] != "2.":
+        session.run(
+            "python",
+            "-c",
+            f"import sys; assert sys.implementation.name == '{implementation}'",
+        )
+
+
+@nox.session(python=["3.7", "3.8", "3.9", "3.10", "pypy3.7", "pypy3.8", "pypy3.9"])
+def github_actions_default_tests(session):
```

Comment:
Could we have a switch in the `noxfile.py` that detects whether it's on GHA (e.g. the `CI` env var) and only includes these sessions in the selected list if we are on CI so that if people just run `nox` locally they don't have a load of cluttered skipped sessions?

We do something similar to skip conda if a user does not have it installed: https://github.com/wntrblm/nox/blob/2ad60058159e78e8a559e4f3a2b50451ea7c295a/noxfile.py#L27-L30

Reply:
I started reworking this in #606 which should probably be reviewed first I guess.